### PR TITLE
Feature/upgrade legacy react dependencies and fix typings

### PR DESCRIPTION
**Pull request by RafaelVidaurre**
**Please make sure the following boxes are checked before submitting your Pull Request:**

- [ ] I've added this PR's link to its Asana task(s)
- [ ] If this PR adds new code that is not going to change in the near future, it includes unit tests to cover it.

### CODE REVIEW

File: packages/new-polymath-ui/src/components/ModalConfirm/ModalConfirm.tsx [hunk 1]
```diff
@@ -32,7 +32,11 @@ const ModalConfirmBase: FC<Props> = props => {
         <Button variant="secondary" onClick={onClose}>
           {cancelButtonText}
         </Button>
-        <ButtonLarge onClick={onSubmit} disabled={isActionDisabled}>
+        <ButtonLarge
+          variant="primary"
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/ModalConfirm/ModalConfirm.tsx]: Why specify the variant if it is one of the default props?

File: packages/new-polymath-ui/src/components/Table/docs/makeData.ts [hunk 2]
```diff
@@ -25,8 +25,8 @@ const newPerson = () => {
   };
 };
 
-export function makeData(...lens) {
-  const makeDataLevel = (depth = 0) => {
+export function makeData(...lens: any[]) {
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/Table/docs/makeData.ts]: Shouldn't `lens` be typed as `number[]`?

File: packages/new-polymath-ui/src/components/Table/docs/makeData.ts [hunk 3]
```diff
@@ -25,8 +25,8 @@ const newPerson = () => {
...
+  const makeDataLevel = (depth = 0): any => {
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/Table/docs/makeData.ts]: Can't this be refined to return the actual data type?

File: packages/new-polymath-ui/src/components/ModalConfirm/ModalConfirm.tsx (hunk 1 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/ModalConfirm/ModalConfirm.tsx]: Yup! I thought I removed all of these

File: packages/new-polymath-ui/src/components/Table/docs/makeData.ts (hunk 2 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/Table/docs/makeData.ts]: Wont fix on this PR this is demo code

File: packages/new-polymath-ui/src/components/Table/docs/makeData.ts (hunk 3 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/Table/docs/makeData.ts]: Wont fix on this PR this is demo code

**Pull request merged**